plugins/lang-plan-a/src/main/java/org/elasticsearch/plan/a/PlanAScriptEngineService.java

```diff
@@ -99,6 +100,10 @@ public Object compile(String script, Map<String, String> params) {
                 // TODO: can we get a real boolean parser in here?
                 compilerSettings.setNumericOverflow(Boolean.parseBoolean(value));
             }
+            value = clone.remove(MAX_LOOP_COUNTER);
```

Comment:
`clone` makes me think of the funky `clone` method. Maybe call it `copy`? Not in this PR, but a thing to think about.